network.go

```diff
@@ -230,13 +278,191 @@ func getLinkByName(netHandle *netlink.Handle, name string, expectedLink netlink.
 		if l, ok := link.(*netlink.Veth); ok {
 			return l, nil
 		}
+	case (&netlink.Macvtap{}).Type():
+		if l, ok := link.(*netlink.Macvtap); ok {
+			return l, nil
+		}
 	default:
 		return nil, fmt.Errorf("Unsupported link type %s", expectedLink.Type())
 	}
 
 	return nil, fmt.Errorf("Incorrect link type %s, expecting %s", link.Type(), expectedLink.Type())
 }
 
+func xconnectVMNetwork(netPair *NetworkInterfacePair, connect bool) error {
+	switch DefaultNetInterworkingModel {
```

Comment:
I think this switch should actually test `netPair.NetInterworkingModel`, and the value assignment of `DefaultNetInterworkingModel` should be done during the `netPair` initialization.

Moreover, I think we should really define a `NetInterConnect` interface with methods `connect` and `disconnect`, with different implementations according to the type of `NetInterworkingModel` in use.

Reply:
@sboeuf I agree with both comments here.

Reply:
@sameo @sboeuf The netpair as we have defined is not the right abstraction for the future. The netpair is only one manifestation. I was planning to rewrite this code to make it generic so it can handle all types of interfaces. So far I have not decided on the right abstraction for the network interface. 

So I would like to defer this comment to a subsequent PR where I would rewrite this code a bit more so that it can support all the upcoming interfaces we plan to support. I will open an issue about the same.

@sboeuf the same is true for our network scanning and caching logic. I will need to change that too to accommodate interfaces other than veth.  If you notice I had to change the scanning logic to do things differently on the second network scan. The second network scan ideally should not be done as it may see results very different from the first one for some use cases.

Reply:
@sameo @sboeuf I added a new issue to track this comment

https://github.com/containers/virtcontainers/issues/420

I need to rewrite a bit of the logic to create a fully qualified endpoint with interfaces vs the current struct.